### ISSUE TITLE
dynamic proxy `NoSuchMethodError` for inherited interface in commons-configuration2

### ISSUE THREAD

**Issue by sify21** [oracle/graal#3256]
**Describe the issue**
I made a [sample project](https://github.com/sify21/test-graal-native) with commons-configuration2. The native-image is built successfully but gives error when running. The build configurations are generated by the agent.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. **git clone https://github.com/sify21/test-graal-native.git**
2. **cd test-graal-native&&./gradlew clean build**
3. **native-image -jar build/libs/Main-1.0-SNAPSHOT.jar Main**
4. **./Main /path/to/any.properties**

**Describe GraalVM and your environment:**
 - GraalVM version: **CE 21.0.0.2**
 - JDK major version: **11**
 - OS: **Ubuntu 16.04.4 LTS**
 - Architecture: **AMD64**

**More details**
Consider adding the `--native-image-info` and  `--verbose` flags when building your native image and paste output below.
<details>

```
Apply jar:file:///home/sexyman/Main-1.0-SNAPSHOT.jar!/META-INF/native-image/native-image.properties
Apply jar:file:///home/sexyman/Main-1.0-SNAPSHOT.jar!/META-INF/native-image/jni-config.json
Apply jar:file:///home/sexyman/Main-1.0-SNAPSHOT.jar!/META-INF/native-image/reflect-config.json
Apply jar:file:///home/sexyman/Main-1.0-SNAPSHOT.jar!/META-INF/native-image/resource-config.json
Apply jar:file:///home/sexyman/Main-1.0-SNAPSHOT.jar!/META-INF/native-image/proxy-config.json
Apply jar:file:///home/sexyman/Main-1.0-SNAPSHOT.jar!/META-INF/native-image/serialization-config.json
Executing [
/home/sexyman/graalvm-ce-java11-21.0.0.2/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-Dsubstratevm.IgnoreGraalVersionCheck=true \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.sparc=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.sparc=ALL-UNNAMED \
--add-exports=org.graalvm.truffle/com.oracle.truffle.api=ALL-UNNAMED \
--add-opens=jdk.internal.vm.compiler/org.graalvm.compiler.debug=ALL-UNNAMED \
--add-opens=jdk.internal.vm.compiler/org.graalvm.compiler.nodes=ALL-UNNAMED \
--add-opens=jdk.unsupported/sun.reflect=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.module=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.ref=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.reflect=ALL-UNNAMED \
--add-opens=java.base/java.io=ALL-UNNAMED \
--add-opens=java.base/java.lang=ALL-UNNAMED \
--add-opens=java.base/java.lang.reflect=ALL-UNNAMED \
--add-opens=java.base/java.lang.invoke=ALL-UNNAMED \
--add-opens=java.base/java.lang.ref=ALL-UNNAMED \
--add-opens=java.base/java.net=ALL-UNNAMED \
--add-opens=java.base/java.nio=ALL-UNNAMED \
--add-opens=java.base/java.nio.file=ALL-UNNAMED \
--add-opens=java.base/java.security=ALL-UNNAMED \
--add-opens=java.base/javax.crypto=ALL-UNNAMED \
--add-opens=java.base/java.util=ALL-UNNAMED \
--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED \
--add-opens=java.base/sun.security.x509=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.logger=ALL-UNNAMED \
--add-opens=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED \
--add-opens=org.graalvm.sdk/org.graalvm.polyglot=ALL-UNNAMED \
--add-opens=org.graalvm.truffle/com.oracle.truffle.polyglot=ALL-UNNAMED \
--add-opens=org.graalvm.truffle/com.oracle.truffle.api.impl=ALL-UNNAMED \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-Xms1g \
-Xmx13460081864 \
-Duser.country=US \
-Duser.language=en \
-Djava.awt.headless=true \
-Dorg.graalvm.version=21.0.0.2 \
-Dorg.graalvm.config= \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Dawt.toolkit=sun.awt.X11.XToolkit \
-Djava.awt.graphicsenv=sun.awt.X11GraphicsEnvironment \
-Djava.awt.printerjob=sun.print.PSPrinterJob \
-Xshare:off \
--module-path \
/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/truffle/truffle-api.jar \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-javaagent:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/svm.jar \
-cp \
/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/llvm-platform-specific-shadowed.jar:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/pointsto.jar:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/javacpp-shadowed.jar:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/llvm-wrapper-shadowed.jar:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/svm.jar:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/svm-llvm.jar:/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/builder/objectfile.jar \
'com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus' \
-watchpid \
32575 \
-imagecp \
/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/library-support.jar:/home/sexyman/Main-1.0-SNAPSHOT.jar \
-H:Path=/home/sexyman \
-H:+DumpTargetInfo \
-H:Class=Main \
-H:FallbackThreshold=0 \
-H:+AllowIncompleteClasspath \
-H:EnableURLProtocols=http \
-H:EnableURLProtocols=https \
-H:+EnableAllSecurityServices \
-H:+ReportUnsupportedElementsAtRuntime \
-H:JNIConfigurationResources=META-INF/native-image/jni-config.json \
-H:ReflectionConfigurationResources=META-INF/native-image/reflect-config.json \
-H:ResourceConfigurationResources=META-INF/native-image/resource-config.json \
-H:DynamicProxyConfigurationResources=META-INF/native-image/proxy-config.json \
-H:SerializationConfigurationResources=META-INF/native-image/serialization-config.json \
-H:CLibraryPath=/home/sexyman/graalvm-ce-java11-21.0.0.2/lib/svm/clibraries/linux-amd64 \
-H:Name=/home/sexyman/Main \

]
[/home/sexyman/Main:32595]    classlist:   1,540.11 ms,  0.96 GB
[/home/sexyman/Main:32595]        (cap):     589.76 ms,  0.96 GB
[/home/sexyman/Main:32595]        setup:   1,969.15 ms,  0.96 GB
# Building image for target platform: org.graalvm.nativeimage.Platform$LINUX_AMD64
# Using native toolchain:
#   Name: GNU project C and C++ compiler (gcc)
#   Vendor: linux
#   Version: 5.4.0
#   Target architecture: x86_64
#   Path: /usr/bin/gcc
# Using CLibrary: com.oracle.svm.core.posix.linux.libc.GLibC
WARNING: Could not register reflection metadata for org.apache.commons.logging.impl.Log4JLogger. Reason: java.lang.NoClassDefFoundError: org/apache/log4j/Priority.
WARNING: Could not register reflection metadata for org.apache.commons.logging.impl.Log4JLogger. Reason: java.lang.NoClassDefFoundError: org/apache/log4j/Priority.
WARNING: Could not register reflection metadata for org.apache.commons.logging.impl.Log4JLogger. Reason: java.lang.NoClassDefFoundError: org/apache/log4j/Priority.
WARNING: Could not register reflection metadata for org.apache.commons.logging.impl.Log4JLogger. Reason: java.lang.NoClassDefFoundError: org/apache/log4j/Priority.
WARNING: Could not register reflection metadata for org.apache.commons.logging.impl.Log4JLogger. Reason: java.lang.NoClassDefFoundError: org/apache/log4j/Priority.
WARNING: Could not register reflection metadata for org.apache.commons.logging.impl.Log4JLogger. Reason: java.lang.NoClassDefFoundError: org/apache/log4j/Priority.
[/home/sexyman/Main:32595]     (clinit):     502.93 ms,  2.31 GB
# Static libraries:
#   graalvm-ce-java11-21.0.0.2/lib/svm/clibraries/linux-amd64/liblibchelper.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libawt_headless.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libawt.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libnet.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libextnet.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libnio.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libjava.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libfdlibm.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libsunec.a
#   graalvm-ce-java11-21.0.0.2/lib/static/linux-amd64/glibc/libzip.a
#   graalvm-ce-java11-21.0.0.2/lib/svm/clibraries/linux-amd64/libjvm.a
# Other libraries: stdc++,stdc++,m,pthread,dl,z,rt
[/home/sexyman/Main:32595]   (typeflow):  11,524.87 ms,  2.31 GB
[/home/sexyman/Main:32595]    (objects):  13,466.12 ms,  2.31 GB
[/home/sexyman/Main:32595]   (features):   1,229.01 ms,  2.31 GB
[/home/sexyman/Main:32595]     analysis:  27,392.39 ms,  2.31 GB
[/home/sexyman/Main:32595]     universe:   1,056.34 ms,  2.31 GB
[/home/sexyman/Main:32595]      (parse):   2,846.24 ms,  3.68 GB
[/home/sexyman/Main:32595]     (inline):   1,652.48 ms,  4.24 GB
[/home/sexyman/Main:32595]    (compile):  12,048.65 ms,  5.54 GB
[/home/sexyman/Main:32595]      compile:  18,391.74 ms,  5.54 GB
[/home/sexyman/Main:32595]        image:   3,336.07 ms,  5.54 GB
[/home/sexyman/Main:32595]        write:     757.02 ms,  5.54 GB
[/home/sexyman/Main:32595]      [total]:  54,639.32 ms,  5.54 GB
```

</details>

Add any other information about the problem here. Especially important are stack traces or log output. Feel free to link to gists or to screenshots if necessary.
```
Exception in thread "main" java.lang.NoSuchMethodError: org.apache.commons.configuration2.builder.fluent.FileBasedBuilderParameters.setFileSystem(org.apache.commons.configuration2.io.FileSystem)
	at com.sun.proxy.$Proxy131.<clinit>(Unknown Source)
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:375)
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.initialize(ClassInitializationInfo.java:295)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1022)
	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1008)
	at org.apache.commons.configuration2.builder.fluent.Parameters.createParametersProxy(Parameters.java:294)
	at org.apache.commons.configuration2.builder.fluent.Parameters.fileBased(Parameters.java:185)
	at Main.main(Main.java:19)
```

**Comment by oubidar-Abderrahim** [comment]
Hi, did you use the agent to generate config files?

**Comment by sify21** [comment]
yes, I have put the configs in the resources folder.

**Comment by sify21** [comment]
Hi @oubidar-Abderrahim , is this a bug instead of a problem with the build configurations?

**Comment by srnagar** [comment]
@gradinac any update on this? I am running into a similar issue.

**Comment by gradinac** [comment]
Hey @sify21, @srnagar, thank you for reporting this issue! I didn't have a chance to look into it, however, I suspect this may already be fixed on master with 2c7eeca64d978fa8d86f2077b74045702834d0b9 - could you try building the image with the latest GraalVM CE dev build from https://github.com/graalvm/graalvm-ce-dev-builds/releases and see if the error still happens?

**Comment by srnagar** [comment]
@gradinac I tried with the latest release a couple of days ago. I'll give it another try with this commit [2c7eeca](https://github.com/oracle/graal/commit/2c7eeca64d978fa8d86f2077b74045702834d0b9)

**Comment by srnagar** [comment]
@gradinac The latest dev build [21.3.0-dev-20210723_1948](https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/21.3.0-dev-20210723_1948) worked for me. I no longer see the above exception.

**Comment by gradinac** [comment]
@srnagar Thank you very much for testing this out! :) Unfortunately, this fix did not make it into 21.2 - it will be in 21.3 that will be released towards the end of October (https://www.graalvm.org/release-notes/version-roadmap/ - Oct 19, 2021: 21.3.0)
I'll close this issue - if the problem reappears, please re-open it.
